### DISCUSSION TITLE
Copy

### DISCUSSION THREAD

**Discussion by giscus** [ct-js/docs.ctjs.rocks#107] [Pages' discussions]
# Copy

Copies are the entities that interact with each other on the screen and what drives the game's logic. They derive from Pixi's AnimatedSprite class, so you can find much more par...

https://docs.ctjs.rocks/Copy.html

**Comment by maddlesp** [comment]
If you are looking for PointDirection (or pdn) it's in Ct.u

**Comment by stoufa** [comment]
Is there a way to assign an id to a copy then query it by that id?
I'm trying to add invisible (opacity = 0) copies in the scene then add a tween to show them at certain point during the game, this should be easier than my current workflow: placing the copy in the room to get its coordinates (x, y, scale.x, scale.y, ..) then remove it and recreate it programmatically.
Thanks in advance.

**Reply by CosmoMyzrailGorynych** [reply]
Usually one would instead create an array in, say, the current room (`ct.room` or `rooms.current` in v4), and walk over it with `for (const copy of rooms.current.yourArray) { ... }`

In room start:
```js
this.transparentCopies = [];
```
In copies' creation:
```js
rooms.current.transparentCopies.push(this);
```
In copies' destruction:
```js
// Removes the copy from the array so you don't try to transition non-existing copies
rooms.current.transparentCopies.splice(rooms.current.transparentCopies.indexOf(this), 1);
```
When you need to tween:
```js
for (const copy of rooms.current.transparentCopies) {
   tween.add({ // ct.tween.add instead of just tween.add for v3
       ...
   });
}
```
Also note that you have `templates.list['YourTemplateName']` if you need to loop over copies of a specific template.

**Reply by stoufa** [reply]
Thank you @CosmoMyzrailGorynych, I will definitely try this out! 😊
